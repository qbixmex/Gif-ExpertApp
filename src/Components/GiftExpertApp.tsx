import { useState } from 'react';
import AddCategory from './AddCategory';
import { GifGrid } from './GifGrid';

type GiftExpertAppProps = {
  defaultCategories?: string[];
};

const GiftExpertApp = ({ defaultCategories = []}: GiftExpertAppProps) => {
  const [categories, setCategories] = useState<string[]>( defaultCategories );

  return (
    <>
      <h1 id="title" className="text-center green">Gif Expert App</h1>

      <AddCategory setCategories={ setCategories } />

      <hr />

      {categories.map(category => (
        <GifGrid
          key={ category }
          category={ category }
        />
      ))}
    </>
  );
};

export default GiftExpertApp;
