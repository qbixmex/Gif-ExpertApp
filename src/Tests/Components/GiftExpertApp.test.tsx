import { shallow } from 'enzyme';
import GiftExpertApp from '../../Components/GiftExpertApp';

// @ts-ignore
describe("Test on GifExpertApp", () => {
  const wrapper = shallow( <GiftExpertApp /> );

  test("Should match with snapshot", () => {
    expect( wrapper ).toMatchSnapshot();
  });
  
  test("Should have a title inside <h1> tag", () => {
    const text = "Gif Expert App";
    expect( wrapper.find("#title").text() ).toBe( text );
  });

  test("Should has <AddCategory /> Component", () => {
    expect( wrapper.find("AddCategory").length ).toBe( 1 );
  });

  test('Should show categories list', () => {
    const categories: string[] = ["Dagon ball", "Shingeki no kyojin"];
    const wrapper = shallow( <GiftExpertApp defaultCategories={ categories } /> );
    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find("GifGrid").length ).toBe( categories.length );
  });
});

