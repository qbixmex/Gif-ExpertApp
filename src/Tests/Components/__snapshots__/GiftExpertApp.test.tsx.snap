// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GifExpertApp Should match with snapshot 1`] = `
<Fragment>
  <h1
    className="text-center green"
    id="title"
  >
    Gif Expert App
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;

exports[`Test on GifExpertApp Should show categories list 1`] = `
<Fragment>
  <h1
    className="text-center green"
    id="title"
  >
    Gif Expert App
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="Dagon ball"
    key="Dagon ball"
  />
  <GifGrid
    category="Shingeki no kyojin"
    key="Shingeki no kyojin"
  />
</Fragment>
`;
